### PR TITLE
Falcon script watch

### PR DESCRIPTION
**Pull request by tutok**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
- [x] `d.ts` watch - resolve https://github.com/deity-io/falcon/issues/514
- [ ] `cjs` watch 

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
No

**Other information:**
No

### CODE REVIEW

File: packages/falcon-logger/src/index.ts [hunk 1]
```diff
@@ -58,16 +56,16 @@ logger.getFor = (moduleName: string): Logger => logger.child({ module: moduleNam
  * @param fn Function to trace the execution time of
  * @returns `fn` result
  */
-logger.traceTime = function<T = any>(label: string, fn: () => Promise<T>): Promise<T> {
```
[the remark this answers is not in the document]
Comment (tutok) [reply packages/falcon-logger/src/index.ts]: ahh, I see, thanks, reverted

File: yarn.lock [hunk 2]
```diff
@@ -16426,6 +16450,13 @@ resolve@^1.11.0:
   dependencies:
     path-parse "^1.0.6"
 
+resolve@^1.11.1:
```
[the remark this answers is not in the document]
Comment (tutok) [reply yarn.lock]: done, also added https://github.com/deity-io/falcon/issues/561

File: packages/falcon-logger/src/index.ts [hunk 3]
```diff
@@ -58,16 +56,16 @@ logger.getFor = (moduleName: string): Logger => logger.child({ module: moduleNam
...
+logger.traceTime = function traceTime<T = any>(label: string, fn: () => Promise<T>): Promise<T> {
   // using `function()` statement to preserve the context in case of "getFor" call
-  if (!this.isLevelEnabled('trace')) {
+  if (!logger.isLevelEnabled('trace')) {
```
Comment (Borales) [inline packages/falcon-logger/src/index.ts]:
```suggestion
  if (!this.isLevelEnabled('trace')) {
```

File: packages/falcon-logger/src/index.ts [hunk 4]
```diff
@@ -58,16 +56,16 @@ logger.getFor = (moduleName: string): Logger => logger.child({ module: moduleNam
...
     return fn();
   }
 
   const startTime = Date.now();
   return fn().finally(() => {
     const duration = Date.now() - startTime;
-    this.trace(`${label} (${duration}ms)`);
+    logger.trace(`${label} (${duration}ms)`);
```
Comment (Borales) [inline packages/falcon-logger/src/index.ts]:
```suggestion
    this.trace(`${label} (${duration}ms)`);
```

File: packages/falcon-logger/src/index.ts (hunk 3 above)
Comment (tutok) [reply packages/falcon-logger/src/index.ts]: done

File: packages/falcon-logger/src/index.ts (hunk 4 above)
Comment (tutok) [reply packages/falcon-logger/src/index.ts]: done

**Pull request merged**